apps/server/src/modules/board/service/board-do.service.ts

```diff
@@ -1,10 +1,11 @@
 import { BadRequestException, Injectable } from '@nestjs/common';
 import { AnyBoardDo } from '@shared/domain';
 import { BoardDoRepo } from '../repo';
+import { DeleteHookService } from './delete-hook.service';
 
 @Injectable()
 export class BoardDoService {
-	constructor(private readonly boardDoRepo: BoardDoRepo) {}
+	constructor(private readonly boardDoRepo: BoardDoRepo, private readonly deleteHookService: DeleteHookService) {}
```

Comment:
My gut feeling is that injecting a deleteHookService to be mandatory feels wrong. On the other hand I know it would have a terrible impact to forget e.g. to inject it and have e.g. orphaned files. ...

Reply:
The idea was: If the service should suppport a delete hook then you have to inject the service. And that's what we do here. The implementation of the service then defines what should actually happen when an object was deleted.